### ISSUE TITLE
mistral-server with n>1 only returns one result

### ISSUE THREAD

**Issue by mmoskal** [EricLBuehler/mistral.rs#955]
## Describe the bug

When passing n=3 to the `/v1/completions` endpoint, I think 3 responses are generated (looking at the token usage and speed) but only one is returned.

I hit /v1/completions with:

```json
{
  "model": "model",
  "stream": false,
  "prompt": "Here is a one line joke: ",
  "max_tokens": 100,
  "temperature": 0.8,
  "n": 3
}
```

I get:

```json
{
  "id": "4",
  "choices": [
    {
      "finish_reason": "stop",
      "index": 2,
      "text": "Why did the computer go to school? Because it wanted to improve its processor's education!\n\nThis joke plays on the double meaning of \"processor\" (a part of a computer that performs calculations, and education, which is what a person goes to school for). It humorously suggests that the computer itself desires to become more knowledgeable, much like a human might.",
      "logprobs": null
    }
  ],
  "created": 1733076506,
  "model": "microsoft/Phi-3.5-mini-instruct",
  "system_fingerprint": "local",
  "object": "text_completion",
  "usage": {
    "completion_tokens": 283,
    "prompt_tokens": 27,
    "total_tokens": 310,
    "avg_tok_per_sec": 8.749153,
    "avg_prompt_tok_per_sec": 35.714287,
    "avg_compl_tok_per_sec": 8.161264,
    "total_time_sec": 35.432,
    "total_prompt_time_sec": 0.756,
    "total_completion_time_sec": 34.676
  }
}
```

```bash
cargo build --release --features metal
./target/release/mistralrs-server --throughput --port 3001 -n 1000  plain -m microsoft/Phi-3.5-mini-instruct -a phi3
```

## Latest commit or version

This is with 09d2e2596d9b0aa728749cf5614e40dd1a112b3b

With current master 3d7e86ef3 I get the same behavior, but output is garbage (will file separate issue)

**Comment by EricLBuehler** [comment]
@mmoskal thanks for the report! I just merged #959 which fixes this on my machine, if you could confirm that would be great.

**Comment by mmoskal** [comment]
Works now, thanks!
